File: Problem6/balance.java
class Solution {
    boolean flag;
    int find_depth(TreeNode root)
    {
        if(root==null)
              return 0;
        int left_depth = find_depth(root.left);
        int right_depth = find_depth(root.right);
        if(Math.abs(left_depth-right_depth)>1)
              flag=false;
        return 1 + Math.max(left_depth,right_depth);
    }
    public boolean isBalanced(TreeNode root) {
        flag = true;
        find_depth(root);
        return flag;
    }
}

File: Problem6/balance.py
from typing import Optional
class Solution:
    flag=True
    def find_depth(root)->int :
        if root is None:
            return 0
        left_depth=Solution.find_depth(root.left)
        right_depth=Solution.find_depth(root.right)
        if abs(left_depth-right_depth)>1 :
             Solution.flag=False
        return 1+max(left_depth,right_depth)
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        Solution.flag=True
        Solution.find_depth(root)
        return Solution.flag